scripts/merge_users.py

```diff
@@ -0,0 +1,285 @@
+
+    mongo_config['host'] = (host or cli_config.get('host') or 'localhost')
+    mongo_config['port'] = (port or cli_config.get('port') or 27017)
+    mongo_config['username'] = username or cli_config.get('username')
+    mongo_config['password'] = password or cli_config.get('password')
+    mongo_config['authdb'] = authdb or cli_config.get('authdb') or mongo_config['mongodb']
+    mongo_config['omim_api_key'] = cli_config.get('omim_api_key')
+
+
+    # always dryrun for now
+    dryrun = True
+    if live:
+        dryrun = False
+
+    LOG.info("Setting database name to %s", mongo_config['mongodb'])
+    LOG.debug("Setting host to %s", mongo_config['host'])
+    LOG.debug("Setting port to %s", mongo_config['port'])
+
+    valid_connection = check_connection(
+        host=mongo_config['host'],
+        port=mongo_config['port'],
+        username=mongo_config['username'],
+        password=mongo_config['password'],
+        authdb=mongo_config['authdb'],
+    )
+
+    LOG.info("Test if mongod is running")
+    if not valid_connection:
+        LOG.warning("Connection could not be established")
+        context.abort()
+
+    try:
+        client = get_connection(**mongo_config)
+    except ConnectionFailure:
+        context.abort()
+
+    database = client[mongo_config['mongodb']]
+
+    LOG.info("Setting up a mongo adapter")
+    mongo_config['client'] = client
+    adapter = MongoAdapter(database)
+
+    user_requests = []
+    event_requests = []
+    acmg_requests = []
+    clinvar_requests = []
+    clinvar_submission_requests = []
+
+    for oi_user_obj in adapter.user_collection.find({'_id': {'$type':'objectId'}}):
+        if not oi_user_obj.get('email'):
+            continue
+
+        LOG.info('===USER===')
+        oi_user_id = oi_user_obj.get('_id')
+        oi_user_email = oi_user_obj.get('email')
+        LOG.info("user: {}".format(oi_user_obj))
+
+        create_alt = False
+        alt_user_obj = adapter.user_collection.find_one({'_id': oi_user_email})
+        if not alt_user_obj:
+            create_alt = True
+            alt_user_obj = copy.deepcopy(oi_user_obj)
+            alt_user_obj['_id'] = oi_user_email
+        else:
+            LOG.info("alt user: {}".format(alt_user_obj))
+            merged_institutes = set()
+            merged_institutes.update(alt_user_obj.get('institutes') + oi_user_obj.get('institutes'))
+            LOG.info('merged institutes: {}'.format(merged_institutes))
+            alt_user_obj['institutes'] = list(merged_institutes)
+
+            merged_roles = set()
+            merged_roles.update(alt_user_obj.get('roles') + oi_user_obj.get('roles'))
+            LOG.info('merged roles: {}'.format(merged_roles))
+            alt_user_obj['roles'] = list(merged_roles)
+
+            created_at = oi_user_obj.get('created_at')
+            alt_created_at = alt_user_obj.get('created_at')
+            if (alt_created_at and alt_created_at < created_at):
```

Comment:
```suggestion
            if ((alt_created_at and created_at) and alt_created_at < created_at):
```

Reply:
Since there is a check for existence of created_at afterwards, and None is smaller than all valid alt_created_at I dont think it matters - but also won't really hurt, so adding!

Reply:
```
In [5]: from datetime import datetime

In [6]: a = datetime.now()

In [7]: a < None
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-cfa6709df3e1> in <module>()
----> 1 a < None

TypeError: unorderable types: datetime.datetime() < NoneType()
```

Reply:
I stand corrected! 🥇